### DISCUSSION TITLE
Custom Widgets

### DISCUSSION THREAD

**Discussion by kendellfab** [fyne-io/fyne#3602] [General]
I've been working on building custom Widgets and have run into a question I haven't found an answer to (to be fair, I could dig into the code, but I hope to get a quicker answer here). For the renderer, I'm trying to understand the order the MinSize and layout are called.

Specifically I'm working on a basic charting library for a project I'm working on. I need to be able to scale the y-axis layout, and I'm unsure of the best place to compute the scaling factor. Any insight would be much appreciated.

**Comment by andydotxyz** [comment]
The two should probably not be connected. The MinSize should be the actual minimum of the chart. You scale the content in Layout according to the size available.

**Comment by kendellfab** [comment]
OK, thanks. I'm definitely working on trying to get  a handle of the best practices for all of this.
